astropy/modeling/tests/test_quantities_parameters.py

```diff
@@ -0,0 +1,476 @@
+    """
+    Basic test of comparison operations on properties with units.
+    """
+
+    g = Gaussian1D(1 * u.J, 1 * u.m, 0.1 * u.m)
+
+    # Essentially here we are checking that parameters behave like Quantity
+
+    assert g.mean == 1 * u.m
+    assert 1 * u.m == g.mean
+    assert g.mean != 1
+    assert 1 != g.mean
+
+    assert g.mean < 2 * u.m
+    assert 2 * u.m > g.mean
+
+    with pytest.raises(UnitsError) as exc:
+        g.mean < 2
+    assert exc.value.args[0] == ("Can only apply 'less' function to "
+                                 "dimensionless quantities when other argument "
+                                 "is not a quantity (unless the latter is all "
+                                 "zero/infinity/nan)")
+
+    with pytest.raises(UnitsError) as exc:
+        2 > g.mean
+    assert exc.value.args[0] == ("Can only apply 'less' function to "
+                                 "dimensionless quantities when other argument "
+                                 "is not a quantity (unless the latter is all "
+                                 "zero/infinity/nan)")
+
+    g = Gaussian1D([1, 2] * u.J, [1, 2] * u.m, [0.1, 0.2] * u.m)
+
+    assert g.mean == [1, 2] * u.m
+    assert np.all([1, 2] * u.m == g.mean)
+    assert g.mean != [1, 2]
+    assert np.all([1, 2] != g.mean)
+
+    with pytest.raises(UnitsError) as exc:
+        g.mean < [3, 4]
+    assert exc.value.args[0] == ("Can only apply 'less' function to "
+                                 "dimensionless quantities when other argument "
+                                 "is not a quantity (unless the latter is all "
+                                 "zero/infinity/nan)")
+
+    with pytest.raises(UnitsError) as exc:
+        [3, 4] > g.mean
+    assert exc.value.args[0] == ("Can only apply 'less' function to "
+                                 "dimensionless quantities when other argument "
+                                 "is not a quantity (unless the latter is all "
+                                 "zero/infinity/nan)")
+
+
+def test_parameter_unit_linking_gaussian():
+    """
+    For some models, units of parameters should be related/consistent. We check
+    the case of the Gaussian here, where the mean and stddev should have the
+    same units.
+    """
+
+    with pytest.raises(UnitsError) as exc:
+        g = Gaussian1D(1 * u.J, 1. * u.m, 0.1 * u.s)
+    assert exc.value.args[0] == ("Parameters 'mean' and 'stddev' should "
+                                 "have matching units")
+
+
+@pytest.mark.xfail
+def test_parameter_unit_linking_custom():
+
+    # TODO: this is just an idea for an API, and open to feedback on the best
+    # way to do this. The current approach is to provide a list of tuples,
+    # where each tuple is a group of units that should have linked units. I
+    # found it easier conceptually to keep this outside the parameter
+    # definition, otherwise we have to navigate all parameters and construct a
+    # tree.
+
```

Comment:
Isn't one way of doing it simply trying to evaluate the model once with the given parameters and trapping errors due to inconsistent units, rather than requiring explicit linkages? This isn't always practical (e.g, if evaluation is expensive), but I tend to lean towards the minimalist specification if that will work. True, it won't necessarily explain what the relationship should be, but sometimes those aren't all that simple anyway. For example, polynomial models are kind of a mess with units, aren't they?